client/my-sites/checklist/checklist-show/index.jsx

```diff
@@ -38,49 +41,57 @@ class ChecklistShow extends PureComponent {
 		} );
 	};
 
-	handleToggle = id => {
-		const { notify, siteId, tasks, update } = this.props;
-		const task = find( tasks, { id } );
+	handleTaskDismiss = task => () => {
+		const { notify, siteId, update } = this.props;
 
 		if ( task && ! task.completed ) {
 			notify( 'is-success', 'You completed a task!' );
-			update( siteId, id );
+			update( siteId, task.id );
 		}
 	};
 
 	render() {
-		const { siteId, tasks } = this.props;
+		const { isJetpack, siteId, taskStatuses } = this.props;
+		const tasks = isJetpack ? jetpackTasks : wpcomTasks;
 
 		return (
 			<Fragment>
 				{ siteId && <QuerySiteChecklist siteId={ siteId } /> }
-				<Checklist
-					isLoading={ ! tasks }
-					tasks={ tasks }
-					onAction={ this.handleAction }
-					onToggle={ this.handleToggle }
-				/>
+				<Checklist isPlaceholder={ ! taskStatuses }>
+					{ tasks.map( task => (
+						<Task
+							buttonPrimary={ task.buttonPrimary }
+							buttonText={ task.buttonText }
+							completed={ task.completed || get( taskStatuses, [ task.id, 'completed' ], false ) }
```

Comment:
Could update individual `tasks`' `completed` attr in `mapStateToProps`. Would Kinda centralize data gathering, but I don't feel strongly.

Reply:
One of the main goals here is to avoid conflation of structure and state. This happens to be one point of overlap, where a task my "structurally" (where the Task is described) declare itself completed, overriding Redux state.

I'd considered trying to get this into `mapStateToProps`, but I dropped that due to some drawbacks I anticipated.
- We're using an object containing state that matches on ID
- As long as our reducers and selectors behave well, objects maintain referential equality and we save rerenders
- Ensuring object equality for objects "created" in `mapStateToProps` is possible, but requires added complexity for little gain
- Alternatively, we could flatten the object. This is appealing because we don't need to worry about equality, but the state doesn't lend itself well to this at this point: `{ [taskId]: { completed: true } }`. The state representation is expected to change, but not in this PR 😉. Additionally, I'd expect to add some complexity to map to/from the flattened task status prop, like `taskStatus${ taskId }`.

For now, let's leave it unless you have input on these thoughts 🙂